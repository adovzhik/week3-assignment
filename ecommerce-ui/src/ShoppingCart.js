import React from 'react';
import PropTypes from 'prop-types';
import Airbnb from './Airbnb.js';

export default class ShoppingCart extends React.Component {

  static propTypes = {
    items: PropTypes.array,
    onRemoveFromCart: PropTypes.func
  }

  render() {
    const items = this.props.items
    .map((item, idx) => {
      return (
        <Airbnb
        title={item.title} 
        houseType={item.houseType}
        image={item.image}
        city={item.location.city}
        country={item.location.country}
        cost={item.payment.cost}
        onClick={this.props.onRemoveFromCart}
        buttonText="Remove from cart"
        buttonVarient="danger"
        description={item.payment.description}
        isInCart={true}
        idx={idx}
        key={idx}
      />
      );
    });

    const total = this.props.items.reduce((previousValue, currentValue) => {
      return previousValue + currentValue.payment.cost;
      }, 0);

      let innerCart;
        if (total > 0) {
          innerCart = <div className='price'>{items.length} Items <br />Total: ${total}</div>
        } else {
          innerCart = <div>Shopping Cart is empty!</div>
        }

      return (
        <div>
          <h2>Shopping Cart</h2>
          <div className='ShoppingCart'>
            {items}
            {innerCart}
          </div>
        </div>
      );
    }
}