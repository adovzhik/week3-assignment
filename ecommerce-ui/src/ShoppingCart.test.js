import React from 'react';
import ShoppingCart from './ShoppingCart';
import { shallow } from 'enzyme';

const testAirbnb = {
  "title": "Denver apartment outside Union Station",
  "houseType": "Apartment",
  "image": "https://images.unsplash.com/photo-1519424187720-db6d0fc5a5d2?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=19f783ae564d7c6da4777cde1a84c1c9&auto=format&fit=crop&w=500&q=80",
  "location": {
    "city": "Denver",
    "country": "USA"
  },
  "payment": {
    "cost": 130,
    "description": ""
  },
  "host": {
    "name": "Jamie",
    "isSuperhost": true
  },
  "rating": {
    "stars": 4.5,
    "reviews": 381
  }
};

it('shallow renders ShoppingCart', () => {
  const wrapper = shallow(
    <ShoppingCart
      items={[testAirbnb]}
      onRemoveFromCart={() => { }}
    />
  );
  expect(wrapper).toMatchSnapshot();
});