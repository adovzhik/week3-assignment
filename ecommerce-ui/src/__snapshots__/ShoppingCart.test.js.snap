// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow renders ShoppingCart 1`] = `
<div>
  <h2>
    Shopping Cart
  </h2>
  <div
    className="ShoppingCart"
  >
    <Airbnb
      buttonText="Remove from cart"
      buttonVarient="danger"
      city="Denver"
      cost={130}
      country="USA"
      description=""
      houseType="Apartment"
      idx={0}
      image="https://images.unsplash.com/photo-1519424187720-db6d0fc5a5d2?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=19f783ae564d7c6da4777cde1a84c1c9&auto=format&fit=crop&w=500&q=80"
      isInCart={true}
      key="0"
      onClick={[Function]}
      title="Denver apartment outside Union Station"
    />
    <div
      className="price"
    >
      1
       Items 
      <br />
      Total: $
      130
    </div>
  </div>
</div>
`;
